src/sdam/server.ts

```diff
@@ -346,11 +347,11 @@ export class Server extends TypedEventEmitter<ServerEvents> {
       ) {
         await this.pool.reauthenticate(conn);
```

Comment:
seems like a miss for CSOT too - should we apply the signal to reauthentication?

Reply:
Circling back here, it wouldn't have been a great experience to not return control to the user when the signal aborts, the same would've been true for auto-connect. I've updated both to have the `await` interrupted without stopping the actual operation (tests are added too).